src/cases/containers/case-hearings/case-hearings.component.ts

```diff
@@ -119,19 +125,21 @@ export class CaseHearingsComponent implements OnInit, OnDestroy {
   }
 
   public createHearingRequest(): void {
-    const hearingCondition: HearingConditions = {
-      mode: Mode.CREATE,
-      isInit: true,
-      caseId: this.caseId
-    };
     this.hearingStore.dispatch(new fromHearingStore.LoadHearingValues(this.caseId));
-    this.hearingStore.dispatch(new fromHearingStore.SaveHearingConditions(hearingCondition));
-    this.router.navigate(['/', 'hearings', 'request']).then();
-  }
-
-  public ngOnDestroy(): void {
-    if (this.sub) {
-      this.sub.unsubscribe();
-    }
+    this.sub = this.hearingStore.select(fromHearingStore.getHearingValuesLastError).subscribe(
+      error => {
+        if (error) {
+          this.router.navigate(['/', 'hearings', 'error']);
+        } else {
+          const hearingCondition: HearingConditions = {
+            mode: Mode.CREATE,
+            isInit: true,
+            caseId: this.caseId
+          };
+          this.hearingStore.dispatch(new fromHearingStore.SaveHearingConditions(hearingCondition));
+          this.router.navigate(['/', 'hearings', 'request']);
+        }
+      }
+    );
```

Comment:
this.hearingStore.dispatch(new fromHearingStore.SaveHearingConditions(hearingCondition)); is not calling the api, so it can be in a separted async block as it is now, as this condition will be reset/reinit in other journey when it's started.
And the API error is coming from the fromHearingStore.LoadHearingValues(this.caseId) only so idealy the redirect could be put there in the effect.

Reply:
I have done the changes as you suggested. The code looks much better and simple now.